pymatgen/phonon/plotter.py

```diff
@@ -590,7 +619,7 @@ def get_ticks(self) -> dict[str, list]:
 
     def plot_compare(
         self,
-        other_plotter: PhononBSPlotter,
+        other_plotter: PhononBSPlotter | Sequence[PhononBSPlotter],
```

Comment:
how about we use `dict[str, PhononBSPlotter]` here so we can label other band structures? would be great to add a legend mapping colors to dict keys.

[see `pymatviz` for what that might look like](https://github.com/janosh/pymatviz/blob/49588d1cce3186322fae1a7c6d4af0211dbfedb2/pymatviz/phonons.py#L108)
![phonon-bands-mp-2758](https://github.com/materialsproject/pymatgen/assets/30958850/02f5ac98-fe2e-4ba9-ba16-211a8875d835)

Reply:
There is also `labels` a few lines below, doesn't that do the same? I could refactor though.

Reply:
yeah i think refactoring would make for a better API here. `labels` was only added 3 months in https://github.com/materialsproject/pymatgen/commit/9268942bb28076c53761882effd4cdb2ba65ec79 ago so i think even a breaking change would be acceptable

Reply:
I removed `labels`. I changed the logic so that no legend is shown if the Plotter is initialised without a label (i.e. if the "original" PhononBS has no label).

Reply:
it's important to still support passing in single `PhononBSPlotters`. made some changes to that effect in 
26fb878. also i think if there are multiple structures, we should always show a legend

Reply:
I added a test for the single `PhononBSPlottter` case.

Reply:
@janosh Is there a best way to add custom colours for the comparison? With `PhononDOSPlotter`, it is done via `add_dos()`, but here it is not obvious to me. Is it better to add the colour to the init or passing multiple colours to `plot_compare`?

Reply:
definitely `plot_compare`. usually it's best for arguments to be as close to where they're used as possible

Reply:
I added a `colors` variable to `plot_compare()`.